### PR TITLE
[gitlab-config] Include other entry points in gitlab ci tasks (trigger jobs) - ACIX 325

### PR DESCRIPTION
**Pull request by CelianR**
### What does this PR do?

This adds a way to parse all gitlab ci configuration entry points from a main entry point (`.gitlab-ci.yml`).
It allows linting separately files included within trigger jobs.

In addition, an LRU cache has been added to improve performance if a file has to be parsed multiple time with the same configuration (same git ref etc.)

This modifies:
- CI Lint
- [CI Diff](https://github.com/DataDog/datadog-agent/pull/27535#issuecomment-2225480913)

### Describe how to test/QA your changes

To show an added entry point, add this job to one of the current gitlab ci files:

```yaml
job:
  stage: notify
  trigger:
    include: .gitlab/a.yml
```

Then add a job in `.gitlab/a.yml`:

```yaml
hello:
  script: echo hello
```

You should see `.gitlab-ci.yml` as modified and `.gitlab/a.yml` as added.

### CODE REVIEW

**Comment by CelianR** [comment]
## Gitlab CI Configuration Changes

<details><summary><h2>.gitlab-ci.yml</h2></summary>

### Modified Jobs
<details>
<summary><b>variables</b></summary>

```diff
  variables:
    AGENT_BINARIES_DIR: bin/agent
    AGENT_QA_PROFILE_SSM_NAME: ci.datadog-agent.agent-qa-profile
    API_KEY_DDDEV_SSM_NAME: ci.datadog-agent.datadog_api_key
    API_KEY_ORG2_SSM_NAME: ci.datadog-agent.datadog_api_key_org2
    APP_KEY_ORG2_SSM_NAME: ci.datadog-agent.datadog_app_key_org2
    ARTIFACTORY_BYPASS_SSM_NAME: ci.datadog-agent.artifactory_bypass
    ARTIFACTORY_GEMS_PATH: artifactory/api/gems/agent-gems
    ARTIFACTORY_PYPI_PATH: artifactory/api/pypi/agent-pypi/simple
    ARTIFACTORY_TOKEN_SSM_NAME: ci.datadog-agent.artifactory_token
    ARTIFACTORY_URL: datadog.jfrog.io
    ARTIFACTORY_USERNAME: datadog-agent
    ARTIFACT_DOWNLOAD_ATTEMPTS: 2
    BTFHUB_ARCHIVE_BRANCH: main
    BUCKET_BRANCH: dev
    CHANGELOG_COMMIT_SHA_SSM_NAME: ci.datadog-agent.gitlab_changelog_commit_sha
    CHOCOLATEY_API_KEY_SSM_NAME: ci.datadog-agent.chocolatey_api_key
    CLANG_LLVM_VER: 12.0.1
    CLUSTER_AGENT_BINARIES_DIR: bin/datadog-cluster-agent
    CLUSTER_AGENT_CLOUDFOUNDRY_BINARIES_DIR: bin/datadog-cluster-agent-cloudfoundry
    CODECOV_TOKEN_SSM_NAME: ci.datadog-agent.codecov_token
    CWS_INSTRUMENTATION_BINARIES_DIR: bin/cws-instrumentation
    DATADOG_AGENT_ARMBUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_ARMBUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_BTF_GEN_BUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_BTF_GEN_BUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_BUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_BUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_EMBEDDED_PATH: /opt/datadog-agent/embedded
    DATADOG_AGENT_SYSPROBE_BUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_WINBUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_WINBUILDIMAGES_SUFFIX: ''
    DD_AGENT_TESTING_DIR: $CI_PROJECT_DIR/test/kitchen
    DD_PKG_VERSION: latest
    DEB_GPG_KEY_ID: c0962c7d
    DEB_GPG_KEY_NAME: Datadog, Inc. APT key
    DEB_GPG_KEY_SSM_NAME: ci.datadog-agent.deb_signing_private_key_${DEB_GPG_KEY_ID}
    DEB_RPM_TESTING_BUCKET_BRANCH: testing
    DEB_S3_BUCKET: apt.datad0g.com
    DEB_SIGNING_PASSPHRASE_SSM_NAME: ci.datadog-agent.deb_signing_key_passphrase_${DEB_GPG_KEY_ID}
    DEB_TESTING_S3_BUCKET: apttesting.datad0g.com
    DOCKER_REGISTRY_LOGIN_SSM_KEY: ci.datadog-agent.docker_hub_login
    DOCKER_REGISTRY_PWD_SSM_KEY: ci.datadog-agent.docker_hub_pwd
    DOCKER_REGISTRY_URL: docker.io
    DOGSTATSD_BINARIES_DIR: bin/dogstatsd
    E2E_TESTS_API_KEY_SSM_NAME: ci.datadog-agent.e2e_tests_api_key
    E2E_TESTS_APP_KEY_SSM_NAME: ci.datadog-agent.e2e_tests_app_key
    E2E_TESTS_RC_KEY_SSM_NAME: ci.datadog-agent.e2e_tests_rc_key
    EXECUTOR_JOB_SECTION_ATTEMPTS: 2
    FF_SCRIPT_SECTIONS: 1
    GENERAL_ARTIFACTS_CACHE_BUCKET_URL: https://dd-agent-omnibus.s3.amazonaws.com
    GET_SOURCES_ATTEMPTS: 2
+   GITHUB_PR_COMMENTER_APP_KEY_SSM_NAME: pr-commenter.github_app_key
+   GITHUB_PR_COMMENTER_INSTALLATION_ID_SSM_NAME: pr-commenter.github_installation_id
+   GITHUB_PR_COMMENTER_INTEGRATION_ID_SSM_NAME: pr-commenter.github_integration_id
    GITLAB_FULL_API_TOKEN_SSM_NAME: ci.datadog-agent.gitlab_full_api_token
    GITLAB_READ_API_TOKEN_SSM_NAME: ci.datadog-agent.gitlab_read_api_token
    GITLAB_SCHEDULER_TOKEN_SSM_NAME: ci.datadog-agent.gitlab_pipelines_scheduler_token
    GO_TEST_SKIP_FLAKE: 'true'
    INSTALL_SCRIPT_API_KEY_SSM_NAME: ci.agent-linux-install-script.datadog_api_key
    INTEGRATION_WHEELS_CACHE_BUCKET: dd-agent-omnibus
    JIRA_READ_API_TOKEN_SSM_NAME: ci.datadog-agent.jira_read_api_token
    KERNEL_MATRIX_TESTING_ARM_AMI_ID: ami-021f04c00ecfa8590
    KERNEL_MATRIX_TESTING_X86_AMI_ID: ami-0c54d42f8f4180b0c
    KITCHEN_AZURE_CLIENT_ID_SSM_NAME: ci.datadog-agent.azure_kitchen_client_id
    KITCHEN_AZURE_CLIENT_SECRET_SSM_NAME: ci.datadog-agent.azure_kitchen_client_secret
    KITCHEN_AZURE_SUBSCRIPTION_ID_SSM_NAME: ci.datadog-agent.azure_kitchen_subscription_id
    KITCHEN_AZURE_TENANT_ID_SSM_NAME: ci.datadog-agent.azure_kitchen_tenant_id
    KITCHEN_EC2_SSH_KEY_SSM_NAME: ci.datadog-agent.aws_ec2_kitchen_ssh_key
    KITCHEN_INFRASTRUCTURE_FLAKES_RETRY: 2
    MACOS_GITHUB_APP_ID_2_SSM_NAME: ci.datadog-agent.macos_github_app_id_2
    MACOS_GITHUB_APP_ID_SSM_NAME: ci.datadog-agent.macos_github_app_id
    MACOS_GITHUB_INSTALLATION_ID_2_SSM_NAME: ci.datadog-agent.macos_github_installation_id_2
    MACOS_GITHUB_INSTALLATION_ID_SSM_NAME: ci.datadog-agent.macos_github_installation_id
    MACOS_GITHUB_KEY_2_SSM_NAME: ci.datadog-agent.macos_github_key_b64_2
    MACOS_GITHUB_KEY_SSM_NAME: ci.datadog-agent.macos_github_key_b64
    MACOS_S3_BUCKET: dd-agent-macostesting
    OMNIBUS_BASE_DIR: /omnibus
    OMNIBUS_GIT_CACHE_DIR: /tmp/omnibus-git-cache
    OMNIBUS_PACKAGE_DIR: $CI_PROJECT_DIR/omnibus/pkg/
    OMNIBUS_PACKAGE_DIR_SUSE: $CI_PROJECT_DIR/omnibus/suse/pkg
    PROCESS_S3_BUCKET: datad0g-process-agent
    RELEASE_VERSION_6: nightly
    RELEASE_VERSION_7: nightly-a7
    RESTORE_CACHE_ATTEMPTS: 2
    RPM_GPG_KEY_ID: b01082d3
    RPM_GPG_KEY_NAME: Datadog, Inc. RPM key
    RPM_GPG_KEY_SSM_NAME: ci.datadog-agent.rpm_signing_private_key_${RPM_GPG_KEY_ID}
    RPM_S3_BUCKET: yum.datad0g.com
    RPM_SIGNING_PASSPHRASE_SSM_NAME: ci.datadog-agent.rpm_signing_key_passphrase_${RPM_GPG_KEY_ID}
    RPM_TESTING_S3_BUCKET: yumtesting.datad0g.com
    RUN_E2E_TESTS: auto
    RUN_KMT_TESTS: auto
    RUN_UNIT_TESTS: auto
    S3_ARTIFACTS_URI: s3://dd-ci-artefacts-build-stable/$CI_PROJECT_NAME/$CI_PIPELINE_ID
    S3_CP_CMD: aws s3 cp $S3_CP_OPTIONS
    S3_CP_OPTIONS: --no-progress --region us-east-1 --sse AES256
    S3_DD_AGENT_OMNIBUS_BTFS_URI: s3://dd-agent-omnibus/btfs
    S3_DD_AGENT_OMNIBUS_LLVM_URI: s3://dd-agent-omnibus/llvm
    S3_DSD6_URI: s3://dsd6-staging
    S3_OMNIBUS_CACHE_BUCKET: dd-ci-datadog-agent-omnibus-cache-build-stable
    S3_PERMANENT_ARTIFACTS_URI: s3://dd-ci-persistent-artefacts-build-stable/$CI_PROJECT_NAME
    S3_PROJECT_ARTIFACTS_URI: s3://dd-ci-artefacts-build-stable/$CI_PROJECT_NAME
    S3_RELEASE_ARTIFACTS_URI: s3://dd-release-artifacts/$CI_PROJECT_NAME/$CI_PIPELINE_ID
    S3_RELEASE_INSTALLER_ARTIFACTS_URI: s3://dd-release-artifacts/datadog-installer/$CI_PIPELINE_ID
    S3_SBOM_STORAGE_URI: s3://sbom-root-us1-ddbuild-io/$CI_PROJECT_NAME/$CI_PIPELINE_ID
    SLACK_AGENT_CI_TOKEN_SSM_NAME: ci.datadog-agent.slack_agent_ci_token
    SMP_ACCOUNT_ID_SSM_NAME: ci.datadog-agent.single-machine-performance-account-id
    SMP_AGENT_TEAM_ID_SSM_NAME: ci.datadog-agent.single-machine-performance-agent-team-id
    SMP_API_SSM_NAME: ci.datadog-agent.single-machine-performance-api
    SMP_BOT_ACCESS_KEY_ID_SSM_NAME: ci.datadog-agent.single-machine-performance-bot-access-key-id
    SMP_BOT_ACCESS_KEY_SSM_NAME: ci.datadog-agent.single-machine-performance-bot-access-key
    SSH_KEY_RSA_SSM_NAME: ci.datadog-agent.ssh_key_rsa
    SSH_KEY_SSM_NAME: ci.datadog-agent.ssh_key
    SSH_PUBLIC_KEY_RSA_SSM_NAME: ci.datadog-agent.ssh_public_key_rsa
    STATIC_BINARIES_DIR: bin/static
    SYSTEM_PROBE_BINARIES_DIR: bin/system-probe
    USE_S3_CACHING: --omnibus-s3-cache
    VCPKG_BLOB_SAS_URL_SSM_NAME: ci.datadog-agent-buildimages.vcpkg_blob_sas_url
    WINDOWS_BUILDS_S3_BUCKET: $WIN_S3_BUCKET/builds
    WINDOWS_TESTING_S3_BUCKET_A6: pipelines/A6/$CI_PIPELINE_ID
    WINDOWS_TESTING_S3_BUCKET_A7: pipelines/A7/$CI_PIPELINE_ID
    WINGET_PAT_SSM_NAME: ci.datadog-agent.winget_pat
    WIN_S3_BUCKET: dd-agent-mstesting
```

</details>

### Added Jobs
<details>
<summary><b>notify_gitlab_ci_changes</b></summary>

```yaml
notify_gitlab_ci_changes:
  before_script:
  - git fetch origin main
  - git checkout main
  - git checkout -
  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
  needs: []
  rules:
  - if: $CI_PIPELINE_SOURCE != "push"
    when: never
  - changes:
      compare_to: main
      paths:
      - .gitlab-ci.yml
      - .gitlab/**/*.yml
  script:
  - source /root/.bashrc
  - python3 -m pip install -r tasks/libs/requirements-github.txt
  - 'export GITHUB_KEY_B64=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh $GITHUB_PR_COMMENTER_APP_KEY_SSM_NAME
    | base64)

    export GITHUB_APP_ID=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh $GITHUB_PR_COMMENTER_INTEGRATION_ID_SSM_NAME)

    export GITHUB_INSTALLATION_ID=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh
    $GITHUB_PR_COMMENTER_INSTALLATION_ID_SSM_NAME)

    export GITLAB_TOKEN=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh $GITLAB_FULL_API_TOKEN_SSM_NAME)

    '
  - inv -e notify.gitlab-ci-diff --pr-comment
  stage: notify
  tags:
  - arch:amd64
```

</details>
<details>
<summary><b>trigger_a</b></summary>

```yaml
trigger_a:
  stage: notify
  trigger:
    include: a.yml
```

</details>

### Changes Summary
| Removed | Modified | Added | Renamed |
| ------- | -------- | ----- | ------- |
| 0 | 1 | 2 | 0 |

:information_source: *Diff available in the [job log](https://celian.dev).*

</details>
<details><summary><h2>.gitlab-ci.yml</h2></summary>

### Modified Jobs
<details>
<summary><b>variables</b></summary>

```diff
  variables:
    AGENT_BINARIES_DIR: bin/agent
    AGENT_QA_PROFILE_SSM_NAME: ci.datadog-agent.agent-qa-profile
    API_KEY_DDDEV_SSM_NAME: ci.datadog-agent.datadog_api_key
    API_KEY_ORG2_SSM_NAME: ci.datadog-agent.datadog_api_key_org2
    APP_KEY_ORG2_SSM_NAME: ci.datadog-agent.datadog_app_key_org2
    ARTIFACTORY_BYPASS_SSM_NAME: ci.datadog-agent.artifactory_bypass
    ARTIFACTORY_GEMS_PATH: artifactory/api/gems/agent-gems
    ARTIFACTORY_PYPI_PATH: artifactory/api/pypi/agent-pypi/simple
    ARTIFACTORY_TOKEN_SSM_NAME: ci.datadog-agent.artifactory_token
    ARTIFACTORY_URL: datadog.jfrog.io
    ARTIFACTORY_USERNAME: datadog-agent
    ARTIFACT_DOWNLOAD_ATTEMPTS: 2
    BTFHUB_ARCHIVE_BRANCH: main
    BUCKET_BRANCH: dev
    CHANGELOG_COMMIT_SHA_SSM_NAME: ci.datadog-agent.gitlab_changelog_commit_sha
    CHOCOLATEY_API_KEY_SSM_NAME: ci.datadog-agent.chocolatey_api_key
    CLANG_LLVM_VER: 12.0.1
    CLUSTER_AGENT_BINARIES_DIR: bin/datadog-cluster-agent
    CLUSTER_AGENT_CLOUDFOUNDRY_BINARIES_DIR: bin/datadog-cluster-agent-cloudfoundry
    CODECOV_TOKEN_SSM_NAME: ci.datadog-agent.codecov_token
    CWS_INSTRUMENTATION_BINARIES_DIR: bin/cws-instrumentation
    DATADOG_AGENT_ARMBUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_ARMBUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_BTF_GEN_BUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_BTF_GEN_BUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_BUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_BUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_EMBEDDED_PATH: /opt/datadog-agent/embedded
    DATADOG_AGENT_SYSPROBE_BUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_SYSPROBE_BUILDIMAGES_SUFFIX: ''
    DATADOG_AGENT_WINBUILDIMAGES: v38375519-53773733
    DATADOG_AGENT_WINBUILDIMAGES_SUFFIX: ''
    DD_AGENT_TESTING_DIR: $CI_PROJECT_DIR/test/kitchen
    DD_PKG_VERSION: latest
    DEB_GPG_KEY_ID: c0962c7d
    DEB_GPG_KEY_NAME: Datadog, Inc. APT key
    DEB_GPG_KEY_SSM_NAME: ci.datadog-agent.deb_signing_private_key_${DEB_GPG_KEY_ID}
    DEB_RPM_TESTING_BUCKET_BRANCH: testing
    DEB_S3_BUCKET: apt.datad0g.com
    DEB_SIGNING_PASSPHRASE_SSM_NAME: ci.datadog-agent.deb_signing_key_passphrase_${DEB_GPG_KEY_ID}
    DEB_TESTING_S3_BUCKET: apttesting.datad0g.com
    DOCKER_REGISTRY_LOGIN_SSM_KEY: ci.datadog-agent.docker_hub_login
    DOCKER_REGISTRY_PWD_SSM_KEY: ci.datadog-agent.docker_hub_pwd
    DOCKER_REGISTRY_URL: docker.io
    DOGSTATSD_BINARIES_DIR: bin/dogstatsd
    E2E_TESTS_API_KEY_SSM_NAME: ci.datadog-agent.e2e_tests_api_key
    E2E_TESTS_APP_KEY_SSM_NAME: ci.datadog-agent.e2e_tests_app_key
    E2E_TESTS_RC_KEY_SSM_NAME: ci.datadog-agent.e2e_tests_rc_key
    EXECUTOR_JOB_SECTION_ATTEMPTS: 2
    FF_SCRIPT_SECTIONS: 1
    GENERAL_ARTIFACTS_CACHE_BUCKET_URL: https://dd-agent-omnibus.s3.amazonaws.com
    GET_SOURCES_ATTEMPTS: 2
+   GITHUB_PR_COMMENTER_APP_KEY_SSM_NAME: pr-commenter.github_app_key
+   GITHUB_PR_COMMENTER_INSTALLATION_ID_SSM_NAME: pr-commenter.github_installation_id
+   GITHUB_PR_COMMENTER_INTEGRATION_ID_SSM_NAME: pr-commenter.github_integration_id
    GITLAB_FULL_API_TOKEN_SSM_NAME: ci.datadog-agent.gitlab_full_api_token
    GITLAB_READ_API_TOKEN_SSM_NAME: ci.datadog-agent.gitlab_read_api_token
    GITLAB_SCHEDULER_TOKEN_SSM_NAME: ci.datadog-agent.gitlab_pipelines_scheduler_token
    GO_TEST_SKIP_FLAKE: 'true'
    INSTALL_SCRIPT_API_KEY_SSM_NAME: ci.agent-linux-install-script.datadog_api_key
    INTEGRATION_WHEELS_CACHE_BUCKET: dd-agent-omnibus
    JIRA_READ_API_TOKEN_SSM_NAME: ci.datadog-agent.jira_read_api_token
    KERNEL_MATRIX_TESTING_ARM_AMI_ID: ami-021f04c00ecfa8590
    KERNEL_MATRIX_TESTING_X86_AMI_ID: ami-0c54d42f8f4180b0c
    KITCHEN_AZURE_CLIENT_ID_SSM_NAME: ci.datadog-agent.azure_kitchen_client_id
    KITCHEN_AZURE_CLIENT_SECRET_SSM_NAME: ci.datadog-agent.azure_kitchen_client_secret
    KITCHEN_AZURE_SUBSCRIPTION_ID_SSM_NAME: ci.datadog-agent.azure_kitchen_subscription_id
    KITCHEN_AZURE_TENANT_ID_SSM_NAME: ci.datadog-agent.azure_kitchen_tenant_id
    KITCHEN_EC2_SSH_KEY_SSM_NAME: ci.datadog-agent.aws_ec2_kitchen_ssh_key
    KITCHEN_INFRASTRUCTURE_FLAKES_RETRY: 2
    MACOS_GITHUB_APP_ID_2_SSM_NAME: ci.datadog-agent.macos_github_app_id_2
    MACOS_GITHUB_APP_ID_SSM_NAME: ci.datadog-agent.macos_github_app_id
    MACOS_GITHUB_INSTALLATION_ID_2_SSM_NAME: ci.datadog-agent.macos_github_installation_id_2
    MACOS_GITHUB_INSTALLATION_ID_SSM_NAME: ci.datadog-agent.macos_github_installation_id
    MACOS_GITHUB_KEY_2_SSM_NAME: ci.datadog-agent.macos_github_key_b64_2
    MACOS_GITHUB_KEY_SSM_NAME: ci.datadog-agent.macos_github_key_b64
    MACOS_S3_BUCKET: dd-agent-macostesting
    OMNIBUS_BASE_DIR: /omnibus
    OMNIBUS_GIT_CACHE_DIR: /tmp/omnibus-git-cache
    OMNIBUS_PACKAGE_DIR: $CI_PROJECT_DIR/omnibus/pkg/
    OMNIBUS_PACKAGE_DIR_SUSE: $CI_PROJECT_DIR/omnibus/suse/pkg
    PROCESS_S3_BUCKET: datad0g-process-agent
    RELEASE_VERSION_6: nightly
    RELEASE_VERSION_7: nightly-a7
    RESTORE_CACHE_ATTEMPTS: 2
    RPM_GPG_KEY_ID: b01082d3
    RPM_GPG_KEY_NAME: Datadog, Inc. RPM key
    RPM_GPG_KEY_SSM_NAME: ci.datadog-agent.rpm_signing_private_key_${RPM_GPG_KEY_ID}
    RPM_S3_BUCKET: yum.datad0g.com
    RPM_SIGNING_PASSPHRASE_SSM_NAME: ci.datadog-agent.rpm_signing_key_passphrase_${RPM_GPG_KEY_ID}
    RPM_TESTING_S3_BUCKET: yumtesting.datad0g.com
    RUN_E2E_TESTS: auto
    RUN_KMT_TESTS: auto
    RUN_UNIT_TESTS: auto
    S3_ARTIFACTS_URI: s3://dd-ci-artefacts-build-stable/$CI_PROJECT_NAME/$CI_PIPELINE_ID
    S3_CP_CMD: aws s3 cp $S3_CP_OPTIONS
    S3_CP_OPTIONS: --no-progress --region us-east-1 --sse AES256
    S3_DD_AGENT_OMNIBUS_BTFS_URI: s3://dd-agent-omnibus/btfs
    S3_DD_AGENT_OMNIBUS_LLVM_URI: s3://dd-agent-omnibus/llvm
    S3_DSD6_URI: s3://dsd6-staging
    S3_OMNIBUS_CACHE_BUCKET: dd-ci-datadog-agent-omnibus-cache-build-stable
    S3_PERMANENT_ARTIFACTS_URI: s3://dd-ci-persistent-artefacts-build-stable/$CI_PROJECT_NAME
    S3_PROJECT_ARTIFACTS_URI: s3://dd-ci-artefacts-build-stable/$CI_PROJECT_NAME
    S3_RELEASE_ARTIFACTS_URI: s3://dd-release-artifacts/$CI_PROJECT_NAME/$CI_PIPELINE_ID
    S3_RELEASE_INSTALLER_ARTIFACTS_URI: s3://dd-release-artifacts/datadog-installer/$CI_PIPELINE_ID
    S3_SBOM_STORAGE_URI: s3://sbom-root-us1-ddbuild-io/$CI_PROJECT_NAME/$CI_PIPELINE_ID
    SLACK_AGENT_CI_TOKEN_SSM_NAME: ci.datadog-agent.slack_agent_ci_token
    SMP_ACCOUNT_ID_SSM_NAME: ci.datadog-agent.single-machine-performance-account-id
    SMP_AGENT_TEAM_ID_SSM_NAME: ci.datadog-agent.single-machine-performance-agent-team-id
    SMP_API_SSM_NAME: ci.datadog-agent.single-machine-performance-api
    SMP_BOT_ACCESS_KEY_ID_SSM_NAME: ci.datadog-agent.single-machine-performance-bot-access-key-id
    SMP_BOT_ACCESS_KEY_SSM_NAME: ci.datadog-agent.single-machine-performance-bot-access-key
    SSH_KEY_RSA_SSM_NAME: ci.datadog-agent.ssh_key_rsa
    SSH_KEY_SSM_NAME: ci.datadog-agent.ssh_key
    SSH_PUBLIC_KEY_RSA_SSM_NAME: ci.datadog-agent.ssh_public_key_rsa
    STATIC_BINARIES_DIR: bin/static
    SYSTEM_PROBE_BINARIES_DIR: bin/system-probe
    USE_S3_CACHING: --omnibus-s3-cache
    VCPKG_BLOB_SAS_URL_SSM_NAME: ci.datadog-agent-buildimages.vcpkg_blob_sas_url
    WINDOWS_BUILDS_S3_BUCKET: $WIN_S3_BUCKET/builds
    WINDOWS_TESTING_S3_BUCKET_A6: pipelines/A6/$CI_PIPELINE_ID
    WINDOWS_TESTING_S3_BUCKET_A7: pipelines/A7/$CI_PIPELINE_ID
    WINGET_PAT_SSM_NAME: ci.datadog-agent.winget_pat
    WIN_S3_BUCKET: dd-agent-mstesting
```

</details>

### Added Jobs
<details>
<summary><b>notify_gitlab_ci_changes</b></summary>

```yaml
notify_gitlab_ci_changes:
  before_script:
  - git fetch origin main
  - git checkout main
  - git checkout -
  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/deb_x64$DATADOG_AGENT_BUILDIMAGES_SUFFIX:$DATADOG_AGENT_BUILDIMAGES
  needs: []
  rules:
  - if: $CI_PIPELINE_SOURCE != "push"
    when: never
  - changes:
      compare_to: main
      paths:
      - .gitlab-ci.yml
      - .gitlab/**/*.yml
  script:
  - source /root/.bashrc
  - python3 -m pip install -r tasks/libs/requirements-github.txt
  - 'export GITHUB_KEY_B64=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh $GITHUB_PR_COMMENTER_APP_KEY_SSM_NAME
    | base64)

    export GITHUB_APP_ID=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh $GITHUB_PR_COMMENTER_INTEGRATION_ID_SSM_NAME)

    export GITHUB_INSTALLATION_ID=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh
    $GITHUB_PR_COMMENTER_INSTALLATION_ID_SSM_NAME)

    export GITLAB_TOKEN=$($CI_PROJECT_DIR/tools/ci/aws_ssm_get_wrapper.sh $GITLAB_FULL_API_TOKEN_SSM_NAME)

    '
  - inv -e notify.gitlab-ci-diff --pr-comment
  stage: notify
  tags:
  - arch:amd64
```

</details>
<details>
<summary><b>trigger_a</b></summary>

```yaml
trigger_a:
  stage: notify
  trigger:
    include: a.yml
```

</details>

### Changes Summary
| Removed | Modified | Added | Renamed |
| ------- | -------- | ----- | ------- |
| 0 | 1 | 2 | 0 |

:information_source: *Diff available in the [job log](https://celian.dev).*

</details>
<details><summary><h2>Added: a.yml</h2></summary>

### Added Jobs
<details>
<summary><b>hello</b></summary>

```yaml
hello:
  script: echo hello
```

</details>

### Changes Summary
| Removed | Modified | Added | Renamed |
| ------- | -------- | ----- | ------- |
| 0 | 0 | 1 | 0 |

:information_source: *Diff available in the [job log](https://celian.dev).*

</details>

**Comment by CelianR** [comment]
## Gitlab CI Configuration Changes

<details><summary><h2>Updated: .gitlab/deploy_containers/deploy_containers_a7.yml</h2></summary>

:warning: Diff too large to display on Github

### Changes Summary
| Removed | Modified | Added | Renamed |
| ------- | -------- | ----- | ------- |
| 0 | 1 | 0 | 0 |

</details>
<details><summary><h2>Updated: .gitlab-ci.yml</h2></summary>

:warning: Diff too large to display on Github

### Changes Summary
| Removed | Modified | Added | Renamed |
| ------- | -------- | ----- | ------- |
| 10 | 121 | 7 | 1 |

</details>

:information_source: *Diff available in the [job log](https://celian.dev).*

File: tasks/linter.py [hunk 1]
```diff
@@ -369,25 +370,57 @@ def is_get_parameter_call(file):
...
+    has_errors = False
+
+    print(f'{color_message("info", Color.BLUE)}: Fetching Gitlab CI configurations...')
+    configs = get_all_gitlab_ci_configurations(ctx)
+
+    def test_gitlab_configuration(entry_point, init_config, context=None):
+        nonlocal has_errors
+
+        # Update config and lint it
+        config = generate_gitlab_full_configuration(ctx, entry_point, context=context, input_config=init_config)
```
Comment (KSerrania) [inline tasks/linter.py:388]: Nit: should we just call it `input_config` everywhere? It seems weird for the same variable to have two different names depending on the method.

File: tasks/libs/ciproviders/gitlab_api.py [hunk 2]
```diff
@@ -462,15 +630,18 @@ def generate_gitlab_full_configuration(
...
+        full_configuration = get_full_gitlab_ci_configuration(ctx, input_file, input_config=input_config)
+    elif input_config:
+        full_configuration = deepcopy(input_config)
     else:
         full_configuration = read_includes(None, input_file, return_config=True)
 
     # Override some variables with a dedicated context
     if context:
-        full_configuration["variables"].update(context)
+        full_configuration['variables'] = full_configuration.get('variables', {}).update(context)
```
Comment (KSerrania) [inline tasks/libs/ciproviders/gitlab_api.py:644]: Why is this change needed?

**Review by KSerrania — COMMENTED** [review]
How did you test? Do you have examples with this change you can mention?

**Comment by CelianR** [comment]
> How did you test? Do you have examples with this change you can mention?

I added a section in "How to test" within the PR description, thanks for the review 👍

File: tasks/libs/ciproviders/gitlab_api.py (hunk 2 above)
Comment (CelianR) [reply tasks/libs/ciproviders/gitlab_api.py:644]: I added this because sometimes there is no `variables` within the configuration (as in the current `deploy_containers_a7.yml`), however this is called only on the main configuration: I'll undo my changes

**Review by pducolin — COMMENTED** [review]
I am not familiar with this functionality, nothing to comment from Agent DevX Loops, would delegate the approval to Agent CI Infra as it seems to be most impacted by the changes

**Review by KSerrania — APPROVED** [review]
